src/components/publishing/sections/view_fullscreen.tsx

```diff
@@ -1,28 +1,39 @@
 import * as PropTypes from "prop-types"
 import * as React from "react"
 import styled from "styled-components"
+import { track } from "../../../utils/track"
 import IconExpand from "../icon/expand"
 
 interface ViewFullscreenProps extends React.HTMLProps<HTMLDivElement> {
   index?: number
 }
 
-const ViewFullscreen: React.SFC<ViewFullscreenProps> = (props, context) => {
-  const onClick = e => {
+@track()
+class ViewFullscreen extends React.Component<ViewFullscreenProps, void> {
+  static childContextTypes = {
+    onViewFullscreen: PropTypes.func,
+  }
+
+  constructor(props) {
+    super()
+    this.onClick = this.onClick.bind(this)
+  }
+
+  @track({ action: "Clicked article impression" })
+  onClick(e) {
```

Comment:
Nit pick, this could be `onClick = (e) => {}` to avoid that `.bind`

Reply:
For some reason I had trouble getting our static analysis tools to not complain using `onClick = () => {}`, but I'll give it another shot and see if I missed something/follow up with a better explanation why.

Reply:
Yeah when using decorators on class-level arrow functions I get `[ts] Unable to resolve signature of property decorator when called as an expression.` in VSCode and `Cannot read property 'configurable' of undefined
TypeError: Cannot read property 'configurable' of undefined` in Storybooks 😒 

There's probably a somewhat reasonable explanation for this with how `onClick = () =>` is assigning a field on the class vs. a method for the "prototype"... but 🤷‍♂️ these edge cases/complexity is what makes me resent all of the OO language feature 🍝  being added to Javascript /rant.

Reply:
Yeah, we've had the same issue in emission - just annoying

Reply:
This is a matter of ES7 features being new still, static properties has only recently reached stage 3.